src/libs/antares/study/scenario-builder/rules.cpp

```diff
@@ -156,25 +156,27 @@ void Rules::readThermalCluster(const AreaName::Vector& splitKey, String value, b
         {
             string clusterId = (area->id).to<string>() + "." + clustername.to<string>();
             disabledClustersOnRuleActive[clusterId].push_back(year);
+            return false;
         }
     }
+    return true;
 }
 
-void Rules::readRenewableCluster(const AreaName::Vector& splitKey, String value, bool updaterMode)
+bool Rules::readRenewableCluster(const AreaName::Vector& splitKey, String value, bool updaterMode)
 {
     const AreaName& areaname = splitKey[1];
     const uint year = splitKey[2].to<uint>();
     const ClusterName& clustername = splitKey[3];
 
     if (not study_.parameters.renewableGeneration.isClusters())
-        return;
+        return false;
```

Comment:
We can keep rules related to renewable clusters even if `isClusters() == false`. I think returning `true` would be better.

Reply:
As the caller of **readLine** function (see **Sets::internalLoadFromINIFile(...)**) does not use **readLine**'s returned status, this little error has no consequence. So I think we should keep the code as it is here.